protocol/v2/qbft/instance/metrics.go

```diff
@@ -24,35 +24,24 @@ func (m *metrics) StartStage() {
 	m.stageStart = time.Now()
 }
 
-func (m *metrics) EndStageProposal(ctx context.Context, round qbft.Round) {
+func (m *metrics) EndStage(ctx context.Context, round qbft.Round, s stage) {
 	validatorStageDurationHistogram.Record(
 		ctx,
 		time.Since(m.stageStart).Seconds(),
 		metric.WithAttributes(
-			stageAttribute(proposalStage),
+			stageAttribute(s),
 			roleAttribute(m.role),
 			observability.DutyRoundAttribute(round)))
 	m.stageStart = time.Now()
 }
 
-func (m *metrics) EndStagePrepare(ctx context.Context, round qbft.Round) {
-	validatorStageDurationHistogram.Record(
```

Comment:
is there a good reason to delete this metric?

Reply:
@moshe-blox its not removed; i replaced the duplicated code with this function which takes stage as an argument

`EndStagePrepare` -> `EndStage(*, stage)`

```go
func (m *metrics) EndStage(ctx context.Context, round qbft.Round, s stage) {
	validatorStageDurationHistogram.Record(
		ctx,
		time.Since(m.stageStart).Seconds(),
		metric.WithAttributes(
			stageAttribute(s),
			roleAttribute(m.role),
			observability.DutyRoundAttribute(round)))
	m.stageStart = time.Now()
}
```